internal/xds/translator/testdata/out/xds-ir/http-upgrade-enable-spdy.routes.yaml

```diff
@@ -0,0 +1,29 @@
+- ignorePortInHostMatching: true
+  name: envoy-gateway/gateway-1/http
+  virtualHosts:
+  - domains:
+    - gateway.envoyproxy.io
+    metadata:
+      filterMetadata:
+        envoy-gateway:
+          resources:
+          - kind: Gateway
+            name: gateway-1
+            namespace: envoy-gateway
+            sectionName: http
+    name: envoy-gateway/gateway-1/http/gateway_envoyproxy_io
+    routes:
+    - match:
+        prefix: /
+      metadata:
+        filterMetadata:
+          envoy-gateway:
+            resources:
+            - kind: HTTPRoute
+              name: httproute-1
+              namespace: default
+      name: httproute/default/httproute-1/rule/0/match/0/gateway_envoyproxy_io
+      route:
+        cluster: httproute/default/httproute-1/rule/0
+        upgradeConfigs:
```

Comment:
websocket is missing when only spdy is set, is that intended ?

Reply:
yes, it's intended.

Reply:
then why would I need the disabled flag ? an empty config can disable websockets

Reply:
An empty slice? will that work?

Reply:
we do have a similar empty value definition in tls.alpnProtocols today 
https://gateway.envoyproxy.io/docs/api/extension_types/#clienttlssettings
cc @guydc

Reply:
I would like to change the API design, keep the style consistency, which mean:
1. remove the disabled know from API
2. if there's nothing configed, use default value(`websocket`)
3. if. there's something configed, adpot it, and you need to explicit `websocket`.
WDYT? cc @envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers

Reply:
> remove the disabled know from API
if there's nothing configed, use default value(websocket)
if. there's something configed, adpot it, and you need to explicit websocket.
WDYT? cc @envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers

LGTM

Reply:
+1 to removing the `disable` field and following a similar APi behavior thats implemented for `alpnProtocols`

Reply:
> +1 to removing the disable field and following a similar APi behavior thats implemented for alpnProtocols

same +1